src/librustc_driver/driver.rs

```diff
@@ -1035,14 +1034,18 @@ pub fn phase_4_translate_to_llvm<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     time(time_passes, "Prepare MIR codegen passes", || {
         let mut passes = ::rustc::mir::transform::Passes::new();
         passes.push_pass(box mir::transform::no_landing_pads::NoLandingPads);
-        passes.push_pass(box mir::transform::remove_dead_blocks::RemoveDeadBlocks);
```

Comment:
I’d rather you kept this separate. As I explained before I can easily think of at least two kinds of passes which would care about dead blocks getting removed but do not need a whole SimplifyCfg machinery to run. Passes like `NoLandingPads` is one of them.  Right-after MIR gets constructed is another (potential interference with analyses). Dataflow passes would be a yet another one.